pkg/api/validation/validation.go

```diff
@@ -1233,6 +1233,28 @@ func validateScaleIOVolumeSource(sio *api.ScaleIOVolumeSource, fldPath *field.Pa
 	return allErrs
 }
 
+func validateScaleIOPersistentVolumeSource(sio *api.ScaleIOPersistentVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if sio.Gateway == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("gateway"), ""))
+	}
+	if sio.System == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("system"), ""))
+	}
+	if sio.VolumeName == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("volumeName"), ""))
+	}
+	if sio.SecretRef != nil {
+		if sio.SecretRef.Name == "" {
```

Comment:
can't tighten validation to make this required without breaking update of existing objects that don't validate this if secretRef is non-nil

Reply:
still outstanding, should not tighten validation on existing objects